cmd/kube-apiserver/app/server.go

```diff
@@ -124,6 +126,11 @@ cluster's shared state through which all other components interact.`,
 
 	fs := cmd.Flags()
 	namedFlagSets := s.Flags()
+	if utilfeature.DefaultFeatureGate.Enabled(zpagesfeatures.ComponentFlagz) {
+		s.Flagz = flagz.NamedFlagSetsReader{
+			FlagSets: namedFlagSets,
```

Comment:
you might need to use `fs` to set the `s.Flagz.FlagSets` since namedFlagSets at this point still holds the default values no?
namedFlagSets are added to the command line flagset on L#138, so might need to this initialization after that

Reply:
Looks like both namedFlagSets and fs have references to the same underlying flag objects, we should be ok here.